step09/README.md

```diff
@@ -0,0 +1,137 @@
+
+```go
+    // process collisions
+    for _, g := range ghosts {
+        if player.position.row == g.position.row && player.position.col == g.position.col {
+            lives = lives - 1
+            if lives != 0 {
+                moveCursor(player.position.row, player.position.col)
+                fmt.Printf(cfg.Death)
+                moveCursor(len(maze)+2, 0)
+                time.Sleep(1000*time.Millisecond) //dramatic pause before reseting player position
+                player.position = player.origin
+            }
+        }
+    }
+```
+
+## Task 03: Update scoreboard to display Player emojis corresponding to number of lives
+
+Previously the number of lives was being displayed as an integer in the game scoreboard. We will now be updating the scoreboard to display the number of lives with player emojis. We will be adding a `getLivesAsEmoji` function to concatenate the correct number of player emojis based on the lives remaining in the game. This function creates a buffer and then writes the player emoji string to the buffer based on the number of lives and then returns that value as a string. This function is called in the last line of the `printScreen` function to update the scoreboard.
+
+```go
+func printScreen() {
+    //...omitted for brevity
+
+    moveCursor(len(maze)+1, 0)
+
+    livesRemaining := strconv.Itoa(lives) //converts lives int to a string
+
+    if cfg.UseEmoji {
+        livesRemaining = getLivesAsEmoji()
+    }
+
+    fmt.Printf("Score: %v\tLives: %v\n", score, livesRemaining)
+}
+
+//concatenate the correct number of player emojis based on lives
+func getLivesAsEmoji() string{
+    buf := bytes.Buffer{}
+    for i := lives; i > 0; i-- {
+        buf.WriteString(cfg.Player)
+    }
+    return buf.String()
+}
+```
+
+So why use a buffer? Turns out there are other ways to concatenate strings in Go. The simplest option just being using the `+` operator to concatenate two strings:
+
+```go
+string1 := "pac"
+string2 : = "go"
+pacgo := string1 + string2 //"pacgo"
+```
+
+For comparison, this is what the `getLivesAsEmoji` function would look like if we used the `+` operator approach.
+
+```go
+func getLivesAsEmoji() string{
+    emojiString := ""
+    for i := lives; i > 0; i-- {
+        emojiString = emojiString + cfg.Player
+    }
+    return emojiString
+}
+```
+
+This version of `getLivesAsEmoji` will be less efficient than the version of the function that uses a buffer. Part of the reason for this performance difference is due to memory allocation required for the string concatentaion. In the version of the function using the `+` operator, there is a memory allocation operation happening for every iteration of the for loop. While for the buffer version of the function there is only a single memory allocation happening when buffer is initiailized. A more detailed example of this performance difference is discussed here https://billglover.me/2019/03/13/learn-go-by-concatenating-strings/
```

Comment:
there is a typo in `to memory allocation required for the string concatentaion`. I would break this paragraph in two separate ones, and mention why there are more allocations - strings in Go are immutable.

It could be as simple as:
```
This version of `getLivesAsEmoji` will be less efficient than the version of the function that uses a buffer. Part of the reason for this performance difference is due to memory allocation required for the string concatenation, as we've seen before that strings in Go are immutable.

In the version of the function using the `+` operator, there is a memory allocation operation happening for every iteration of the for loop. While for the buffer version of the function there is only a single memory allocation happening when buffer is initialised. A more detailed example of this performance difference is discussed [here](https://billglover.me/2019/03/13/learn-go-by-concatenating-strings/)
```

I prefer to keep links hidden with markdown syntax.

Reply:
Ok I updated Readme per your input. Thanks for the help with this piece!

Reply:
Thanks for your contribution!